src/h5web/visualizations/index.test.ts

```diff
@@ -1,559 +1,408 @@
-import {
-  HDF5Collection,
-  HDF5TypeClass,
-  HDF5ShapeClass,
-  HDF5Metadata,
-  HDF5Entity,
-  HDF5Attribute,
-  HDF5LinkClass,
-} from '../providers/models';
+import { HDF5Metadata, HDF5Entity } from '../providers/models';
 import { Vis, VIS_DEFS } from '.';
 import { NxInterpretation } from './nexus/models';
-
-const baseGroup = { id: 'group', collection: HDF5Collection.Groups as const };
-const baseDataset = {
-  id: 'dataset',
-  collection: HDF5Collection.Datasets as const,
-};
-const baseDatatype = {
-  id: 'datatype',
-  collection: HDF5Collection.Datatypes as const,
-};
-
-const nxDataGroup = {
-  ...baseGroup,
-  attributes: [
-    { name: 'NX_class', value: 'NXdata' },
-    { name: 'signal', value: 'my_signal' },
-  ] as HDF5Attribute[],
-  links: [
-    {
-      class: HDF5LinkClass.Hard as const,
-      title: 'my_signal',
-      collection: HDF5Collection.Datasets,
-      id: baseDataset.id,
-    },
-  ],
-};
-
-function supportsEntity(
-  vis: Vis,
-  entity: HDF5Entity,
-  metadata = {} as HDF5Metadata
-): boolean {
-  return VIS_DEFS[vis].supportsEntity(entity, metadata);
+import {
+  makeSimpleDataset,
+  intType,
+  makeGroup,
+  makeDatatype,
+  advancedType,
+  makeDataset,
+  scalarShape,
+  makeSimpleShape,
+  stringType,
+  makeStrAttr,
+  makeMetadata,
+  withAttributes,
+  makeDatasetHardLink,
+  makeNxDataGroup,
+} from '../providers/mock/data';
+
+const datasetIntScalar = makeDataset('dataset_int', intType, scalarShape);
+const datasetFltScalar = makeDataset('dataset_flt', intType, scalarShape);
```

Comment:
`datasetFltScalar` was created with `intType`.

Anyway, there is no distinction (that I know of) between a `datasetIntScalar` and `datasetIntScalar` so perhaps one is sufficient.

Reply:
Good catch. `isBaseType` actually checks for int, float or string, so it makes sense to have a test for it in scalar vis. I'm adding a check with a `datasetStrScalar` for completeness.